### DISCUSSION TITLE
Electron+next.js, how to get hardware information?

### DISCUSSION THREAD

**Discussion by ChangeYourself0613** [community/community#135838] [Programming Help]
### Body

I built desktop app using electron + next.js.
So I made exe app using electron.
But in that app, I wanna get device's hardware information, but don't know how to do.
Plz feel free to reach out to me if you have any experiences.
Thanks

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Youssef-Meziani** [comment]
To get hardware information in an Electron + Next.js app, you can use Node.js modules like _os_ or third-party libraries like _systeminformation_:

**1. Using os:**
```
const os = require('os');
const cpuInfo = os.cpus();
const totalMem = os.totalmem();
```

**2. Using systeminformation:**
```
const si = require('systeminformation');
si.cpu().then(data => console.log(data));
```

Include one of these JavaScript codes in your Electron backend to retrieve hardware details.
